web/build.sh

```diff
@@ -21,558 +21,90 @@ cd "$THIS_SCRIPT_PATH"
-#
-# The block above would copy index.js into the build output folder for app/web's debug
-# product.
-#
-# ``` bash
-#   rm -rf "$(output_path app/web)"
-# ```
-#
-# The block above is useful for deleting all app/web build products as part of a `clean`
-# action.
-#
-# ### Other Notes
-#
-# In the future, we may opt to move $INTERMEDIATE stuff underneath both $DEBUG and $RELEASE,
-# making it a third param.  This is currently unclear, but if so, we'd do
-# $DEBUG/$INTERMEDIATE and $RELEASE/$INTERMEDIATE via a third argument.
-output_path ( ) {
-  if [ $# -lt 1 ]; then
-    builder_die "Insufficient argument count!"
-  elif [ $# -eq 1 ]; then
-    # Used by clean:<target> actions
-    echo "$BUILD_BASE/$1"
-  else
-    echo "$BUILD_BASE/$1/$2"
-  fi
-}
+OUTPUT_DIR=obj
 
 SOURCE="src"
 
-SENTRY_RELEASE_VERSION="release@$VERSION_WITH_TAG"
-
 # Ensures that we rely first upon the local npm-based install of Typescript.
 # (Facilitates automated setup for build agents.)
-PATH="../node_modules/.bin:$PATH"
+PATH="../../../node_modules/.bin:$PATH"
 
 compiler="npm run tsc --"
 compilecmd="$compiler"
 
-PREDICTIVE_TEXT_SOURCE="../common/predictive-text/unit_tests/in_browser/resources/models/simple-trie.js"
-PREDICTIVE_TEXT_OUTPUT="src/test/manual/web/prediction-ui/simple-en-trie.js"
-
-builder_describe "Builds Keyman Engine for Web (KMW)." \
-  "@../common/web/keyman-version build" \
-  "@../common/web/input-processor build" \
-  "@src/tools/building/sourcemap-root build" \
+builder_set_child_base src
+builder_describe "Builds engine modules for Keyman Engine for Web (KMW)." \
   "clean" \
   "configure" \
   "build" \
-  "test             Runs unit tests.  Only $(builder_term test:web) is currently defined"  \
-  ":embed           Builds the configuration of KMW used within the Keyman mobile apps" \
-  ":engine          Builds all common code used by other targets" \
-  ":web             Builds the website-oriented configuration of Keyman Engine for Web" \
-  ":ui              Builds the desktop UI modules used by the $(builder_term :web) build product" \
-  ":samples         Builds only sample & test pages found under src/samples and src/test" \
-  ":tools           Builds related development + unit-test resources" \
-  "--no-minify      Skips any minification steps in the build" \
-  "--all            Sets action to run on KMW's submodules as well if appropriate (actions - $(builder_term test))"
+  "test" \
+  ":engine/configuration     Subset used to configure KMW" \
+  ":engine/device-detect     Subset used for device-detection " \
+  ":engine/dom-utils         A common subset of function used for DOM calculations, layout, etc" \
+  ":engine/element-wrappers  Subset used to integrate with website elements" \
+  ":engine/keyboard-cache    Subset used to collate keyboards and request them from the cloud" \
+  ":engine/main              Builds all common code used by KMW's app/-level targets" \
+  ":engine/osk               Builds the Web OSK module"
```

Comment:
I thought of it as a very welcome feature; this definitely reduces the amount of overhead to set the child targets in place.

Also of note:  while I could change the behavior that I implemented to match, any custom paths for targets will ignore a configured `builder_set_child_base` command.  I'd have to use `:configuration=src/engine/configuration` instead.

Note how long this tends to make the script's lines.  Granted, it's not the _worst_ consequence, but I could see it offending people's linter / review preferences.  I also thought it pretty ugly to have to keep re-repeating the same regular pattern due to heavy child-root structure bias; I'd strongly prefer to utilize the feature, accidental though it may be.  Granted, yes, just a "preference".

The current target-name pattern seen above also is a bit more "documenting" for any incomers to the project - the build script's help itself will document the path to the child project.  Of course, inspection of the script's source _would_ only be one step away.  So... I suppose it's not the _biggest_ difference.  Also, this _is_ either the first or second script to be reworked to make use of the parent-child builder feature... it'd be kind of odd for the "exception" to be more common than the "rule" - for custom paths to be more common than default paths... though I suppose it may not actually be "more common" once Developer's script (at minimum) adopts the feature.

Note that `app/` targets will exist, so I can't just set the base folder to `src/engine`.

Reply:
OK, I guess we can support `/` in target names. We should update the docs accordingly to make this clear.